test/flippant/rules_test.exs

```diff
@@ -2,15 +2,58 @@ defmodule Flippant.RulesTest do
   use ExUnit.Case, async: true
 
   alias Flippant.Rules
+  alias Flippant.Rules.Default
 
-  describe "enabled_for_actor?/3" do
-    actor = %{id: 1, name: "Parker"}
-    group = %{"chosen" => fn %{id: id}, value -> id == value end}
+  describe "enabled_for_actor?/2 with default rules" do
+    test "nobody group is always disabled" do
+      actor = %{id: 1, name: "Parker"}
+      rule = %{"nobody" => []}
 
-    refute Rules.enabled_for_actor?([], actor)
-    refute Rules.enabled_for_actor?([], actor, %{})
-    refute Rules.enabled_for_actor?([{"chosen", 1}], actor, %{})
+      refute Rules.enabled_for_actor?(rule, actor)
+    end
 
-    assert Rules.enabled_for_actor?([{"chosen", 1}], actor, group)
+    test "everybody group is always enabled for a feature" do
+      actor = %{id: 1, name: "Parker"}
+      rule = %{"everybody" => []}
+
+      assert Rules.enabled_for_actor?(rule, actor)
+    end
+
+    test "unknown groups are always disabled for a feature" do
+      actor = %{id: 1, name: "Parker"}
+      rule = %{"users" => [1]}
+
+      refute Rules.enabled_for_actor?(rule, actor)
+    end
+  end
+
+  defmodule TestRules do
+    def enabled?("staff", _enabled_for, %{staff?: true}), do: true
+
+    def enabled?("users", enabled_for, actor) do
+      Enum.any?(enabled_for, &(actor.id == &1))
+    end
+
+    def enabled?(_group, _enabled_for, _actor), do: false
+  end
+
+  describe "enabled_for_actor?/2 with custom rules" do
+    setup do
+      Application.put_env(:flippant, :rules, TestRules)
```

Comment:
This modified the application state so this test shouldn’t be async. Eventually that won’t be an issue, when we no longer use the application env. For now it could cause flaming tests.

Reply:
Done!